### DISCUSSION TITLE
How can I manually prevent a target from being rebuilt?

### DISCUSSION THREAD

**Discussion by Kodiologist** [ropensci/targets#794] [Help] [answered]
Suppose I make a trivial code change to a target and I don't want all the downstream targets to be rebuilt. (Maybe I'll just want one immediately downstream target rebuilt to ensure I got the same results as before, but if I do, I certainly don't need the entire set of downstream targets rebuilt.) How can I do that?

**Comment by wlandau** [comment, marked as answer]
You can suppress those targets with `tar_target(..., cue = tar_cue(mode = "never"))` or comment them out in `_targets.R`.

**Reply by Kodiologist** [reply]
I think I was unclear. I want to keep all the downstream targets. I just want them to treat the currently built values of those targets as up-to-date, instead of treating them as outdated due to a trivial upstream change.

**Reply by Kodiologist** [reply]
In other words, I want to treat a target that actually has changed (trivially) as if it hasn't changed.

**Reply by wlandau** [reply]
`tar_cue(mode = "never")` does that: https://docs.ropensci.org/targets/reference/tar_cue.html

**Reply by Kodiologist** [reply]
Okay, I think I see. That does seem to mean that I need to then remove `tar_cue(mode = "never")` later when the upstream target is changed again, in order for it be able to see dependencies again. Is there a way to tell `targets` that the current built version of a target should be considered up-to-date, but it can rebuild it if the dependencies change again? I guess it would entail replacing its hash of its dependencies without actually rebuilding it.

**Reply by wlandau** [reply]
There are other arguments to `tar_cue()` that customize when a target runs (e.g. `command`, `file`, and `depend`). Otherwise, `targets` decides whether a code change is trivial based on `deparse()`. If it deparses your code and the hash is different, then `targets` will think the code is different. But if you change whitespace or comments, those changes get ignored in deparsing and will not invalidate the target.

**Reply by Kodiologist** [reply]
Yeah, the tricky part is when the code changes are not just whitespace or comment changes but the effect is supposed to be the same, as in the case of refactoring. If it's not possible to just manually replace a target's memory of the old hash with the new hash, then I guess them's the breaks and I should  add and remove `tar_cue(mode = "never")` as necessary instead. Thanks for your help.
